### PR TITLE
merge downdrafts v0 to main clubbmf branch

### PR DESCRIPTION
**Pull request by adamrher**

### CODE REVIEW

File: src/physics/cam/clubb_mf.F90 [hunk 1]
```diff
@@ -323,11 +375,14 @@ subroutine integrate_mf( nz,
...
      !
-     ! fraction of rain detrained into downdrafts
-     real(r8),parameter                   :: fdd = 0._r8
+     ! height here downdrafts feel the surface
+     real(r8),parameter                   :: z00dn = 1.e3_r8
      !
-     ! fixed entrainment rate (debug only)
-     real(r8),parameter                   :: fixent = 2.e-4_r8
+     ! to fix entrainmnet rate
+     logical                              :: fixent = .false.
```
Comment (mchinita) [inline src/physics/cam/clubb_mf.F90:382]: Maybe these logical variables should be on the namelist (e.g., this one, do_condensation, etc). I understand that they are rarely changed but when they do, it requires editing clubb_mf.F90 and compiling the model again, right?

File: src/physics/cam/clubb_mf.F90 [hunk 2]
```diff
@@ -450,10 +545,76 @@ subroutine integrate_mf( nz,
...
+         upu(1,i) = u(1)
+         upv(1,i) = v(1)
+
+         upqt(1,i)  = cwqt * upw(1,i) * sigmaqt/sigmaw
+         upthv(1,i) = cwthv * upw(1,i) * sigmathv/sigmaw
+       enddo
+
+       facqtu=1._r8
+       facthvu=1._r8
+       if (scalesrf) then
```
Comment (mchinita) [inline src/physics/cam/clubb_mf.F90:566]: What is this block for? I saw that it is turned off though (scalesrf =.false.).

File: src/physics/cam/clubb_mf.F90 [hunk 3]
```diff
@@ -450,10 +545,76 @@ subroutine integrate_mf( nz,
...
+         ! scale surface fluxes
+         srfwqtu = 0._r8
+         srfwthvu = 0._r8
+         srfarea = 0._r8
+         do i=1,clubb_mf_nup
+             srfwqtu=srfwqtu+upqt(1,i)*upw(1,i)*upa(1,i)
```
Comment (mchinita) [inline src/physics/cam/clubb_mf.F90:573]: This loops throughout the N updrafts but these variables assume the latest value (they don't have indexes). I don't remember what is this block for though, so ignore my comment if it doesn't make sense!

File: src/physics/cam/clubb_mf.F90 [hunk 4]
```diff
@@ -450,10 +545,76 @@ subroutine integrate_mf( nz,
...
+       enddo
+
+       facqtu=1._r8
+       facthvu=1._r8
+       if (scalesrf) then
+         ! scale surface fluxes
+         srfwqtu = 0._r8
+         srfwthvu = 0._r8
+         srfarea = 0._r8
+         do i=1,clubb_mf_nup
```
Comment (mchinita) [inline src/physics/cam/clubb_mf.F90:572]: This loops throughout the N updrafts but these variables assume the latest value (they don't have indexes). I don't remember what is this block for though, so ignore my comment if it doesn't make sense!

File: src/physics/cam/clubb_mf.F90 (hunk 2 above)
Comment (mchinita) [reply src/physics/cam/clubb_mf.F90:566]: I understand now what this does. General comment: would it be possible to add a few more comments to the code? Throughout the code but also when the variables are defined at the beginning of the script. I understand how "boring" it is, and I'm guilty of doing the same thing (not doing it in this case)... but it really helps others to read the code faster! Thanks!

File: src/physics/cam/clubb_mf.F90 [hunk 5]
```diff
@@ -450,10 +545,76 @@ subroutine integrate_mf( nz,
...
+             srfwthvu=srfwthvu+upthv(1,i)*upw(1,i)*upa(1,i)
+             srfarea = srfarea+upa(1,i)
+         end do
+         facqtu=srfarea*wqt/srfwqtu
+         facthvu=srfarea*wthv/srfwthvu
+       end if
+
+       do i=1,clubb_mf_nup
+
+         betaqt = (qt(4)-qt(2))/(0.5_r8*(dzt(4)+2._r8*dzt(3)+dzt(2)))
```
Comment (mchinita) [inline src/physics/cam/clubb_mf.F90:583]: Is this for the interpolation of qt from the thermodynamic grid to the momentum one?

File: src/physics/cam/clubb_mf.F90 [hunk 6]
```diff
@@ -423,10 +489,35 @@ subroutine integrate_mf( nz,
...
+     dnrr  = 0._r8
+     dnth  = 0._r8
+     dnqc  = 0._r8
+     dnql  = 0._r8
+     dnqi  = 0._r8
+     dnqs  = 0._r8
+     dnlmix= 0._r8
+     sdnqt = 0._r8
+     sdnthl= 0._r8
+     dnw   = 0._r8
```
Comment (mchinita) [inline src/physics/cam/clubb_mf.F90]: dnw was already initialized on L492.

**Comment by adamrher** [comment]
@mikaelwitte @rls347 @mchinita I am going to merge this PR today. If you want to squeeze in a quick review, you have until EOD. No biggie if you can't find the time. I'm sure the code is flawless ;)

File: src/physics/cam/clubb_mf.F90 (hunk 1 above)
Comment (adamrher) [reply src/physics/cam/clubb_mf.F90:382]: yes, you need to re-compile if you change these values. I'm going to punt on making all these namelists, although I do agree they should be.

File: src/physics/cam/clubb_mf.F90 (hunk 2 above)
Comment (adamrher) [reply src/physics/cam/clubb_mf.F90:566]: sure. I can try to make a habit of more comments. For this one in particular, I added "(req'd for conservation if not running with zero-flux b.c.'s)". This functionality is anticipation of a future enhancement where we recognize the area of the plumes and therefore can lift our zero-flux b.c.'s assumption.

File: src/physics/cam/clubb_mf.F90 (hunk 5 above)
Comment (adamrher) [reply src/physics/cam/clubb_mf.F90:583]: yes, it's to get qt and thl at the surface from their values on the thermodynamic grid.

File: src/physics/cam/clubb_mf.F90 (hunk 6 above)
Comment (adamrher) [reply src/physics/cam/clubb_mf.F90]: fixed.

**Pull request merged**